Scripts/RemediationScripts/Remediate-BackupVaultImmutability.ps1

```diff
@@ -0,0 +1,534 @@
+        $AutoRemediation,
+
+        [String]        
+        [Parameter(ParameterSetName = "WetRun", HelpMessage = "Specifies the time of creation of file to be used for logging remediation details when AutoRemediation switch is used")]
+        $TimeStamp
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+
+    if ($PerformPreReqCheck) {
+        try {
+            Write-Host "[Step 1 of 4] Validate and install the modules required to run the script and validate the user..."
+            Write-Host "Setting up prerequisites..." -ForegroundColor $([Constants]::MessageType.Info)    
+            Setup-Prerequisites
+            Write-Host "Completed setting up prerequisites." -ForegroundColor $([Constants]::MessageType.Update)    
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        catch {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::DoubleDashLine)
+            return
+        }
+    }
+    else {
+        Write-Host "[Step 1 of 4] Validate the user..."
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context)) {
+        Write-Host "Connecting to Azure account..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    if (-not($AutoRemediation)) {
+        Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+        Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+        Write-Host "Account Name: [$($context.Account.Id)]"
+        Write-Host "Account Type: [$($context.Account.Type)]"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "To enable and lock immutability for Backup Vault(s) in a Subscription, Owner or higher privileged role assignment on the Subscription is required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "[Step 2 of 4] Fetch all Backup Vault(s)"
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # list to store Backup Vault details.
+    $backupVaultDetails = @()
+
+    # To keep track of remediated and skipped resources    
+    $logRemediatedResources = @()    
+    $logSkippedResources = @()    
+
+    # Control Id    
+    $controlIds = "Azure_BackupVault_DP_Enable_Immutability_Trial"
```

Comment:
remove trail from control and from dependent source

Reply:
Removed Trial tag